### ISSUE TITLE
Follow renames in file history is broken

### ISSUE THREAD

**Issue by FallenGameR** [gitextensions/gitextensions#3215]
I've just build locally latest GitExtensions from code. Looks nice but I found that now file history doesn't follow renames.

Current release 2.48.05:
Settings:
![image](https://cloud.githubusercontent.com/assets/711295/15979294/0bcde2d8-2f18-11e6-8088-5c43865012e1.png)
File history (master branch is selected since otherwise it shows tons of not relevant commits):
![image](https://cloud.githubusercontent.com/assets/711295/15979336/4d3daab4-2f18-11e6-931c-31de39c9883f.png)

Trunk from 6/10/2016:
Settings:
![image](https://cloud.githubusercontent.com/assets/711295/15979345/613959fa-2f18-11e6-8c08-ae323d968f8c.png)
File history (master branch is selected to be on-par with other screenshot):
![image](https://cloud.githubusercontent.com/assets/711295/15979365/8762dde0-2f18-11e6-8b08-77b3ab8e9513.png)

**Comment by FallenGameR** [comment]
Found what fixes it. Here is a patch for it. --simplify-by-decoration option passed to git log drops valid history for some reason.

```
From 88f81051aee4c4add5de8844c205b3ad3404ad67 Mon Sep 17 00:00:00 2001
From: Alexander Kostikov <Alex.Kostikov@gmail.com>
Date: Fri, 10 Jun 2016 15:29:44 -0700
Subject: [PATCH] Fix for file history traversal

---
 GitUI/CommandsDialogs/FormFileHistory.cs | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/GitUI/CommandsDialogs/FormFileHistory.cs b/GitUI/CommandsDialogs/FormFileHistory.cs
index c1e6ad2..8697b10 100644
--- a/GitUI/CommandsDialogs/FormFileHistory.cs
+++ b/GitUI/CommandsDialogs/FormFileHistory.cs
@@ -185,7 +185,7 @@ private FixedFilterTuple BuildFilter(string fileName)
 
             if (AppSettings.FullHistoryInFileHistory)
             {
-                res.RevisionFilter = string.Concat(" --full-history --simplify-by-decoration ", res.RevisionFilter);
+                res.RevisionFilter = string.Concat(" --full-history ", res.RevisionFilter);
             }
 
             res.PathFilter = " \"" + fileName + "\"";
-- 
2.8.2.windows.1
```

**Comment by vbjay** [comment]
https://github.com/gitextensions/gitextensions/issues?utf8=%E2%9C%93&q=rename+history

**Comment by vbjay** [comment]
Good catch.  Might want to post an issue n git repo.

**Comment by FallenGameR** [comment]
>Might want to post an issue on git repo. 

Not sure it is a bug in git. Here is extract from git log help:

> The --simplify-by-decoration option allows you to view only the big picture of the topology of the history, by omitting commits that are not referenced by tags. Commits are marked as !TREESAME (in other words, kept after history simplification rules described above) if (1) they are referenced by tags, or (2) they change the contents of the paths given on the command line. All other commits are marked as TREESAME (subject to be simplified away).

So (2) can be related to this behavior. I'm not familiar with GitExtensions code base. Why this option was added in the first place? It's intent is to hide history partly. This can be ok for command line mode, but GUI tool can definitely handle long history visualization.

**Comment by FallenGameR** [comment]
Commit that introduced this authored by @Radrik5.
Dmitriy, could you review the fix?

**Comment by Radrik5** [comment]
As you can see from my commit https://github.com/gitextensions/gitextensions/commit/d177560958db04f65dba01c1fd491bede9002ba9, I just changed `res.Filter` to `res.RevisionFilter`, so you need to go deeper :)

This branch is executed when Full History is enabled in settings. From your description it seems you don't need full history, so could you please try unchecking it in the context menu of the file history graph?

**Comment by FallenGameR** [comment]
>I just changed res.Filter to res.RevisionFilter, so you need to go deeper :)
Not true, this change was not a simple refactoring:
- res.Filter = string.Concat(" --full-history --simplify-by-decoration ", res.Filter);
+ res.RevisionFilter = string.Concat(" --full-history --simplify-by-decoration ", res.RevisionFilter);

--simplify-by-decoration was added. It was not present in the code before. And this addition breaks the file history with renames.

>This branch is executed when Full History is enabled in settings
This setting is enabled in this case. See the screenshot from the first message in this thread.

>From your description it seems you don't need full history
On the contrary. Full file history is broken in trunk now in case file was renamed.

How contributions work in GitExtensions? Would owners apply the patch or I need to send a PR?

**Comment by Radrik5** [comment]
Initially `--simplify-by-decoration` was added with `--full-history` 5 years ago in commit https://github.com/gitextensions/gitextensions/commit/7d6dd8ca02e832b2ee2a3f55c6b768b7252060ea by @spdr870. It's been in all the releases since 2.27, so it's not something added recently by mistake.

@FallenGameR, I rephrase my question: are you sure you need to have *Full history* enabled in the *context menu*? We don't have this option on the settings page from your screenshots.

![image](https://cloud.githubusercontent.com/assets/9339459/16035720/329e55c6-3221-11e6-97f6-b071f4b2b731.png)

**Comment by FallenGameR** [comment]
> It's been in all the releases since 2.27, so it's not something added recently by mistake.

Don't know about the history of this line, but the latest published release don't have this error as seen by the screenshots. And it doesn't have this switch.

>are you sure you need to have Full history enabled in the context menu? 

Yes, I do have it enabled in this context menu. Just recompiled trunk and double checked. Full history is checked and history is broken at rename commit.

**Comment by FallenGameR** [comment]
Closed in #3250
